### ISSUE TITLE
MenuItemLink does not apply active style in (v4.0 beta 1)

### ISSUE THREAD

**Issue by panfiva** [marmelab/react-admin#7337]
```js
const StyledMenuItemLink = styled(MenuItemLink)(({ theme }) => ({
	'color': theme.palette.text.secondary,
	'&.RaMenuItemLink-active': {
		color: theme.palette.text.primary,
	},
}))

<StyledMenuItemLink to='/posts' primaryText='Posts' leftIcon={<SearchIcon />} />
```

**What you were expecting:**
When I got to List, Show, Create and Edit components for `posts` resource, `.RaMenuItemLink-active` should apply

**What happened instead:**
`.RaMenuItemLink-active` only applies to `List`

**Related code:**
The issue is related to https://github.com/remix-run/react-router/issues/7954

Specifically, the following code in React-admin's `MenuItemLink` component does not return correct results
```js
    const match = useMatch(
        typeof props.to === 'string' ? props.to : props.to.pathname
    );
```

- When in list component (`/posts`), useMatch will return successful match
- When in child components (/posts/1, /posts/1/show, etc), it will return failed match

**Suggested fix**

Add "end:false" attribute to useMatch search. 

```js
const to = (typeof props.to === 'string' ? props.to : props.to.pathname) || ''
const match = useMatch({ path: to, end: false })
```

* React-admin version: 4.0 beta 1

**Comment by WiXSL** [comment]
Thanks for the report

**Comment by fzaninotto** [comment]
Fixed by #7342
